src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/HotResourceSummary.java

```diff
@@ -88,8 +109,31 @@ public void buildSummaryMessageAndAddToFlowUnit(FlowUnitMessage.Builder messageB
 
   public static HotResourceSummary buildHotResourceSummaryFromMessage(
       HotResourceSummaryMessage message) {
-    HotResourceSummary newSummary = new HotResourceSummary(message.getResourceType(),
-        message.getThreshold(), message.getValue(), message.getUnitType(), message.getTimePeriod());
+    String resourceTypeClassName = message.getResourceType();
+    //cast to Enum using reflection. find the last "." and separate the class name into two parts.
+    //e.g. com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.Resources$ResourceType.HEAP
+    //will be split to  "com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.Resources$ResourceType"
+    //and "HEAP"
+    String[] nameSplit = resourceTypeClassName.split("\\.(?=[^\\.]+$)");
```

Comment:
Can we do this without using reflection? I think its ok to have some boiler plate code here. Ideally, we should define objects that are sent over the network as protobuf objects to avoid this scenario.

Reply:
Done. Created a ResourceType protobuf message instead and removed the use of  refection/regex from code.